src/api/callApi.ts

```diff
@@ -1,20 +1,43 @@
 import axios from 'axios';
+import {bookType} from '../screens/HomeScreen';
+import {chapterType} from '../screens/ChapterScreen';
 
-export const getBook = async () => {
-  let response = await axios.post(`https://api-dev.lelivrescolaire.fr/graph`, {
-    query:
-      'query{viewer{books{hits{id displayTitle url subjects{name}levels{name}valid}}}}',
-  });
+interface ApiResponse<T> {
+  data: {
+    viewer: {
+      [key: string]: {
```

Comment:
That's much better. But the key remains untyped. Can you improve that, so that when I get a BookApiResponse, I know that `data.viewer.books.hits` is where the array of `bookType` is, rather than `data.viewer.INCORRECT.hits`.

Reply:
[key in 'books' | 'chapters']